### PR TITLE
Cleanup LoadGameLevel()

### PR DESCRIPTION
**Pull request by kphoenix137**
This PR breaks up `LoadGameLevel()` into several functions, to make it more readable. Originally, this function is obscenely long and hard to follow. I was extremely careful to not rearrange any function calls to avoid introducing any bugs.

### CODE REVIEW

**Review by AJenbo — APPROVED** [review]
Probably all of this should be moved to a different file to avoid all the LoadGame* prefixing. Also I'm not 100% happy about the functions that are named after the bool that is checked to see if it should be called or not. But still this looks like a good start.

**Comment by AJenbo** [comment]
Lol so you didn't actually leave things unchanged.

Please seperat Clean ups and bug fixes, it's hard to review when they are mixed together and it makes it hard to write the Change log later

**Comment by kphoenix137** [comment]
> Lol so you didn't actually leave things unchanged.
> 
> Please seperat Clean ups and bug fixes, it's hard to review when they are mixed together and it makes it hard to write the Change log later 

I'm sorry. I wrote this one up making an assumption the other PR would be merged first. I should have drafted this PR until that one was reviewed.

**Comment by kphoenix137** [comment]
> Lol so you didn't actually leave things unchanged.
> 
> Please seperat Clean ups and bug fixes, it's hard to review when they are mixed together and it makes it hard to write the Change log later 

I'm sorry. I wrote this one up making an assumption the other PR would be merged first. I should have drafted this PR until that one was reviewed.
